include/solvers/navier_stokes_scratch_data.h

```diff
@@ -728,34 +707,60 @@ class NavierStokesScratchData : public PhysicsScratchDataBase
         for (unsigned int j = 0; j < number_of_particles; ++j)
           cell_void_fraction[j] = cell_void_fraction_bulk;
       }
+  }
 
-    if (number_of_particles == 0)
-      return;
-
-
-    // Calculate the average particle velocity within
-    // the cell
-    average_particle_velocity = average_particle_velocity / number_of_particles;
+  /** @brief Creates an object of type Quadrature<dim> that contains the
+   * particle reference location. This object can be used to interpolate any
+   * function known at the dofs at the location of the particles.
+   */
 
+  Quadrature<dim>
+  calculate_particle_location_quadrature_points()
```

Comment:
It's not a calculation but more of a gathering step no?
```suggestion
  gather_particles_reference_location()
```